dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowFactoryImpl.java

```diff
@@ -968,14 +952,14 @@ public void saveAction(final WorkflowAction action) throws DotDataException,Alre
 			db.addParam(action.getIcon());
 			db.addParam(action.isRoleHierarchyForAssign());
 			db.addParam(action.isRequiresCheckout());
-			//db.addParam(action.getRequiresCheckoutOption());
+			db.addParam(WorkflowStatus.toCommaSeparatedString(action.getShowOn()));
 			db.addParam(action.getId());
 			db.loadResult();
 		}
 
-		final WorkflowStep proxyStep = new WorkflowStep();
-		proxyStep.setId(action.getStepId());
-		cache.removeActions(proxyStep);
+		final List<WorkflowStep> relatedProxiesSteps =
+				this.findProxiesSteps(action);
+		relatedProxiesSteps.forEach( proxyStep -> cache.removeActions(proxyStep) );

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Specify a type for: 'proxyStep' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2211)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)